sdk/src/types/block/protocol.rs

```diff
@@ -45,39 +45,41 @@ pub struct ProtocolParameters {
     pub(crate) token_supply: u64,
     /// Genesis timestamp at which the slots start to count.
     #[serde(with = "crate::utils::serde::string")]
-    pub(crate) genesis_unix_timestamp: u32,
+    pub(crate) genesis_unix_timestamp: u64,
     /// Duration of each slot in seconds.
     pub(crate) slot_duration_in_seconds: u8,
     /// The number of slots in an epoch expressed as an exponent of 2.
-    pub(crate) slots_per_epoch_exponent: u32,
+    pub(crate) slots_per_epoch_exponent: u8,
     /// The amount of potential Mana generated by 1 IOTA in 1 slot.
-    pub(crate) mana_generation_rate: u32,
+    pub(crate) mana_generation_rate: u8,
     /// The scaling of `mana_generation_rate` expressed as an exponent of 2.
-    pub(crate) mana_generation_rate_exponent: u32,
+    pub(crate) mana_generation_rate_exponent: u8,
     #[packable(unpack_error_with = |_| Error::InvalidManaDecayFactors)]
     #[getset(skip)]
     /// A lookup table of epoch index diff to mana decay factor.
-    pub(crate) mana_decay_factors: BoxedSlicePrefix<u32, u8>,
+    pub(crate) mana_decay_factors: BoxedSlicePrefix<u32, u16>,
     /// The scaling of `mana_decay_factors` expressed as an exponent of 2.
-    pub(crate) mana_decay_factors_exponent: u32,
+    pub(crate) mana_decay_factors_exponent: u8,
     /// An integer approximation of the sum of decay over epochs.
     pub(crate) mana_decay_factor_epochs_sum: u32,
     /// The scaling of `mana_decay_factor_epochs_sum` expressed as an exponent of 2.
-    pub(crate) mana_decay_factor_epochs_sum_exponent: u32,
+    pub(crate) mana_decay_factor_epochs_sum_exponent: u8,
     /// The unbonding period in epochs before an account can stop staking.
     pub(crate) staking_unbonding_period: EpochIndex,
-    /// TODO
+    /// The slot index used by tip-selection to determine if a block is eligible by evaluating issuing times
+    /// and commitments in its past-cone to accepted tangle time and last committed slot respectively.
```

Comment:
```suggestion
    /// and commitments in its past-cone against accepted tangle time and last committed slot respectively.
```

Sorry, re-reading this I have one more suggestion